src/main/java/net/sf/jabref/logic/cleanup/BiblatexCleanup.java

```diff
@@ -25,33 +26,22 @@
             entry.getField(oldFieldName).ifPresent(oldValue -> {
                 if (!oldValue.isEmpty() && (!entry.getField(newFieldName).isPresent())) {
                     // There is content in the old field and no value in the new, so just copy
-                    entry.setField(newFieldName, oldValue);
-                    changes.add(new FieldChange(entry, newFieldName, null, oldValue));
-
-                    entry.clearField(oldFieldName);
-                    changes.add(new FieldChange(entry, oldFieldName, oldValue, null));
+                    changes.add(entry.setField(newFieldName, oldValue).orElse(null));
+                    changes.add(entry.clearField(oldFieldName).orElse(null));
                 }
             });
         }
-
         // Dates: create date out of year and month, save it and delete old fields
         // If there already exists a value for the field date, it is not overwritten
         String date = "";
         if (entry.getField(FieldName.DATE).isPresent()) {
             date = entry.getField(FieldName.DATE).get();
         }
-        if (date.isEmpty()) {
+        if (StringUtils.isBlank(date)) {
```

Comment:
isBlank also accepts an optional I think (if not, please add a corresponding overload), hence `Optional<String> date = entry.getField(FieldName.DATE)` should work.